src/java/org/apache/cassandra/repair/ParentRepairSessionListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.repair;
+
+import java.util.UUID;
+import java.util.concurrent.Future;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.service.ActiveRepairService;
+
+public interface ParentRepairSessionListener
+{
+    ParentRepairSessionListener instance = CassandraRelevantProperties.REPAIR_PARENT_SESSION_LISTENER.isPresent()
+                                      ? make(CassandraRelevantProperties.REPAIR_PARENT_SESSION_LISTENER.getString())
+                                      : new ParentRepairSessionListener() {};
```

Comment:
I'd rather we explicitly had `NoopParentRepairSessionListener` instead of relying on empty default implementations in the interface.
unless you expect that a typical listener has mostly empty implementation...